### PR TITLE
Mutiple registries query support

### PR DESCRIPTION
**Pull request by GeekArthur**
Signed-off-by: Jingfu Wang <jingfwan@redhat.com>

**Please specify the area for this PR**
Registry library

**What does does this PR do / why we need it**:
This PR adds the support for multiple registries query.

**Which issue(s) this PR fixes**:

Fixes https://github.com/devfile/api/issues/377

**PR acceptance criteria**:

- [x] Test (WIP) 

- [x] Documentation (WIP)

**How to test changes / Special notes to the reviewer**:
1. Build the registry library CLI then connect to multiple registries by running `export REGISTRY_LIST=https://registry.devfile.io/,http://devfile-registry.192.168.64.2.nip.io`
2. Run `./registry-library list --type stack` to list stacks from multiple registries

### CODE REVIEW

**Review by johnmcollier — CHANGES_REQUESTED** [review]
Left some review comments.

File: tests/integration/pkg/tests/devfilelibrary_tests.go [hunk 1]
```diff
@@ -24,30 +24,44 @@ import (
 	"github.com/onsi/gomega"
 )
 
+const (
+	devfilePublicRegistry = "https://registry.devfile.io"
+	devfileStageRegistry  = "https://registry.stage.devfile.io"
```
[the remark this answers is not in the document]
Comment (GeekArthur) [reply tests/integration/pkg/tests/devfilelibrary_tests.go:29]: Sure, will update those to get them from the config.

File: tests/integration/cmd/devfileregistry_test.go [hunk 2]
```diff
@@ -35,13 +36,13 @@ const (
 //SynchronizedBeforeSuite blocks is executed before run all test suites
 var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
 	fmt.Println("Starting to setup objects before run ginkgo suite")
-	registry := os.Getenv("REGISTRY")
-	if registry != "" {
-		config.Registry = registry
-	} else {
-		config.Registry = "https://registry.devfile.io"
-		os.Setenv("REGISTRY", config.Registry)
+	registryList := os.Getenv("REGISTRY_LIST")
```
[the remark this answers is not in the document]
Comment (GeekArthur) [reply tests/integration/cmd/devfileregistry_test.go:39]: Sounds good, will update it.

File: registry-library/library/library.go [hunk 3]
```diff
@@ -110,26 +115,49 @@ func GetRegistryIndex(registry string, devfileTypes ...indexSchema.DevfileType)
...
-	if err != nil {
-		return err
+		registryList = GetMultipleRegistriesIndices(registryURLArray, indexSchema.StackDevfileType, indexSchema.SampleDevfileType)
 	}
 
 	w := tabwriter.NewWriter(os.Stdout, 5, 2, 3, ' ', tabwriter.TabIndent)
-	fmt.Fprintln(w, "Name", "\t", "Description", "\t")
-	for _, stackIndex := range registryIndex {
-		fmt.Fprintln(w, stackIndex.Name, "\t", stackIndex.Description, "\t")
+	fmt.Fprintln(w, "Name", "\t", "Description", "\t", "Registry", "\t", "Error", "\t")
```
[the remark this answers is not in the document]
Comment (GeekArthur) [reply registry-library/library/library.go:152]: It's not used by odo, only the `registry-library` uses it for testing, so no breaking changes for odo.

File: registry-library/library/library.go [hunk 4]
```diff
@@ -110,26 +115,49 @@ func GetRegistryIndex(registry string, devfileTypes ...indexSchema.DevfileType)
 	return registryIndex, nil
 }
 
+// GetMultipleRegistriesIndices returns returns the list of stacks and/or samples of multiple registries
+func GetMultipleRegistriesIndices(registryURLs []string, devfileTypes ...indexSchema.DevfileType) []Registry {
```
[the remark this answers is not in the document]
Comment (GeekArthur) [reply registry-library/library/library.go:119]: > Can you add unit tests for this function if possible?
The integration test uses `registry-library` CLI for testing and the `registry-library` calls the function directly, given that do you think we still need the unit test for this function? IMO, the integration test already covered the unit test as there are no logic/code block between `registry-library` and the function itself.

[the remark this answers is not in the document]
Comment (johnmcollier) [reply registry-library/library/library.go]: I'm fine with that

**Comment by GeekArthur** [comment]
@johnmcollier Comments addressed, please review again.

**Review by johnmcollier — APPROVED** [review]

**Pull request merged**